SU2_CFD/src/solvers/CBFMSolver.cpp

```diff
@@ -0,0 +1,605 @@
+                                      CNumerics **numerics_container, CConfig *config, unsigned short iMesh) 
+{   
+
+}
+
+void CBFMSolver::Source_Template(CGeometry *geometry, CSolver **solver_container, CNumerics *numerics,
+                                                 CConfig *config, unsigned short iMesh) { }
+
+void CBFMSolver::BC_Euler_Wall(CGeometry      *geometry,
+                                    CSolver        **solver_container,
+                                    CNumerics      *conv_numerics,
+                                    CNumerics      *visc_numerics,
+                                    CConfig        *config,
+                                    unsigned short val_marker) { }
+
+void CBFMSolver::BC_HeatFlux_Wall(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config, unsigned short val_marker) { }
+
+void CBFMSolver::BC_Far_Field(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config,
+                                     unsigned short val_marker) { }
+
+void CBFMSolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config,
+                                 unsigned short val_marker) { }
+
+void CBFMSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config,
+                                  unsigned short val_marker) { }
+
+void CBFMSolver::BC_Sym_Plane(CGeometry      *geometry,
+                                   CSolver        **solver_container,
+                                   CNumerics      *conv_numerics,
+                                   CNumerics      *visc_numerics,
+                                   CConfig        *config,
+                                   unsigned short val_marker) { }
+
+void CBFMSolver::BC_Custom(CGeometry *geometry, CSolver **solver_container, CNumerics *conv_numerics, CNumerics *visc_numerics, CConfig *config, unsigned short val_marker) { }
+
+void CBFMSolver::ExplicitRK_Iteration(CGeometry *geometry, CSolver **solver_container,
+                                             CConfig *config, unsigned short iRKStep) { }
+
+void CBFMSolver::ExplicitEuler_Iteration(CGeometry *geometry, CSolver **solver_container, CConfig *config) { }
+
+void CBFMSolver::ImplicitEuler_Iteration(CGeometry *geometry, CSolver **solver_container, CConfig *config) { }
+
+su2double CBFMSolver::ComputeKMach(CSolver **solver_container, unsigned long iPoint, vector<su2double*>W_cyl){
+    su2double Temperature = solver_container[FLOW_SOL]->GetNodes()->GetTemperature(iPoint);
+    su2double SoS = solver_container[FLOW_SOL]->GetNodes()->GetSoundSpeed(iPoint);
+    su2double W_mag{0};
+    for(unsigned short iDim=0; iDim<nDim; ++iDim){
+        W_mag += (*W_cyl.at(iDim))*(*W_cyl.at(iDim));
+    }
+    W_mag = sqrt(W_mag);
```

Comment:
geometry toolbox for these types of operation as well
pass W_cyl by const reference.